### DISCUSSION TITLE
Allow models to pick a new model

### DISCUSSION THREAD

**Discussion by LanceLake** [oobabooga/text-generation-webui#4806] [Ideas]
Sometimes, I want to ask coding questions. I have a model for that. 

Sometimes, I want to RP. I have a model for that.

Sometimes, I want to have it do stock analysis. I have a model for that. 

I would like to be able to put in a description for each model and then have 1 character. This character will look at all the models available and pick the best one to use. Then it will load said model and resubmit the question for a proper answer.

Either that or assign a model to a character. When you load that character, you autoload that model. Then allow for characters to switch characters. "Oh. I don't know this, but StockBot does. Let me ask".

**Comment by mongolu** [comment]
Interesting. What model do you use for stocks analysis? 🙂

**Comment by LanceLake** [comment]
Some Llama2 70b model, but it has it's own specifics of a low temp and so on. I also trained it using superbogav2 and some other stuff. It's not great, but it tends to hit usually.

**Comment by lawyinking** [comment]
thinking about this question too!
